File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    for i in range(1,number +1):
        temp = 0
        for j in range(1,int(i**0.5)+1):
            if i % j == 0:
                temp += 1
                if j ** 2 != i:
                    temp += 1
        if temp <= limit:
            answer += temp
        else:
            answer += power
    return answer

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let weights = [];
    for(let i=1; i<=parseInt(number); i++){
        let temp = [];
        for(let j=1; j<=Math.sqrt(i); j++){
            if(i % j == 0){
                temp.push(j);
                if(i / j !== j){
                    temp.push(i/j);
                }
            }
            
        }
        if(temp.length > limit){
            weights.push(power);
        }
        else{
            weights.push(temp.length);
        }
    }
    let answer = weights.reduce((acc,cur) => acc+cur,0);
    return answer;
}